packages/schema/lib/schemas/AuthFieldSchema.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const makeSchema = require('../utils/makeSchema');
+const BaseFieldSchema = require('./BaseFieldSchema');
+const schema = makeSchema(
+  {
+    id: '/AuthFieldSchema',
+    description: 'Field schema specialized for auth fields (e.g., OAuth).',
+    type: 'object',
+    allOf: [
+      { $ref: BaseFieldSchema.id },
+      {
+        type: 'object',
+        properties: {
+          // New property
+          isSafe: {
+            description: 'Indicates if this auth field is safe (not secret).',
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+      // Forbid certain keys from having isSafe = true
+      {
+        not: {
+          properties: {
+            key: {
+              enum: [
+                'access_token',
+                'refresh_token',
+                'api_key',
+                'password',
+                'secret',
+              ],
+            },
+            isSafe: { const: true },
+          },
+          required: ['key', 'isSafe'],
+        },
+      },
```

Comment:
This is out of scope of this ticket. Meant to be handled [separately](https://zapierorg.atlassian.net/browse/PDE-5720), but it is fine if you prefer to handle it together because it is small.

Reply:
I was thinking of handling the "smaller" field constraints in it makes perfect sense to separate them out 👍

Reply:
Sounds good

Reply:
Feel free to collapse both tickets into 1.